composer/trainer/trainer.py

```diff
@@ -2115,6 +2120,7 @@ def _train_loop(self) -> None:
                     # did (e.g. duration specified in samples/batches/tokens), but it is still
                     # the end of the dataloader (i.e. next(dataloader) would raise StopIteration)
                     if self.state.train_metrics is not None:
+                        self.state.train_metrics = self._ensure_metrics_device_and_dtype(self.state.train_metrics)
```

Comment:
why this change?

Reply:
5 batches per epoch, i checkpoint after batch 5. 

When checkpoint loads, model starts epoch 0 and then hits the end of epoch section because dataloader is exhausted. It tries to compute metric and hits metric is not dense (because we normally do this in the update call, but there were no batches run for epoch 0). 

This was previously not an issue as resuming a checkpoint meant loading metrics which were subsequently moved to GPU in load logic. 

There's a unit test that verifies this behavior in `load_checkpoint.py::test_resumption`